### PR TITLE
added findVariable method to pyminsky class, updated findObject to guarante ...

### PR DESCRIPTION
**Pull request by wdednam**
…e stability, and refactored tests that call findVariable.

- - -
This change is [Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/533)

### CODE REVIEW

**Review by wdednam — COMMENTED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/highperformancecoder/minsky/533)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @wdednam)

File: RESTService/pyminsky.cc [hunk 1]
```diff
@@ -30,28 +30,73 @@ namespace pyminsky
 
   minsky::Item& findObject(const std::string& typeName)
   {
-    using namespace minsky;
-    auto& canvas = minsky.canvas;
+      using namespace minsky;
+      auto& canvas = minsky.canvas;
 
-    canvas.item.reset();
+      canvas.item.reset(); // Reset the item to ensure no leftover references
```
Comment (highperformancecoder) [inline RESTService/pyminsky.cc:36]: indentation has gone a bit weird

File: RESTService/pyminsky.cc [hunk 2]
```diff
@@ -30,28 +30,73 @@ namespace pyminsky
...
 
-    if (typeName == "Group" && !canvas.model->groups.empty())
-      canvas.item = canvas.model->groups.front();
-    else
-      minsky.model->recursiveDo(&GroupItems::items, [&](const Items&, Items::const_iterator i)
+      bool found = false;
```
Comment (highperformancecoder) [inline RESTService/pyminsky.cc:38]: This found flag is redundant. It just shadows the validity status of canvas.item

File: RESTService/pyminsky.cc [hunk 3]
```diff
@@ -30,28 +30,73 @@ namespace pyminsky
...
+                  if (v->name() == name)
+                  {
+                      canvas.item = *i;
+                      return true; // Stop recursion
+                  }
+              }
+              return false;
+          });
+
+      if (!found || !canvas.item)
```
Comment (highperformancecoder) [inline RESTService/pyminsky.cc:90]: found flag redundant here too.

**Review by highperformancecoder — APPROVED** [review]

File: RESTService/pyminsky.cc (hunk 1 above)
Comment (wdednam) [reply RESTService/pyminsky.cc:36]: Oops. I must have accidentally pressed the tab key on that line. I'll check the indentation more carefully prior to a pull request next time.

File: RESTService/pyminsky.cc (hunk 2 above)
Comment (wdednam) [reply RESTService/pyminsky.cc:38]: Understood. I was getting a segfault in the renameVariables.sh test, so I used it for added safety.

**Pull request merged**